### PR TITLE
adpd188: initial commit

### PR DESCRIPTION
**Pull request by adrimbarean**
Add adpd188 python class and examples on how to use it.

Signed-off-by: Andrei Drimbarean <Andrei.Drimbarean@analog.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Tested with hardware on RPI 3 and 4 system

**Test Configuration**:
* Hardware: RPI 3, RPI 4
* OS: Linux, Windows

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tfcollins** [comment]
Need a doc entry

**Comment by adrimbarean** [comment]
V1:
 - added doc entry;

File: adi/adpd188.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        # self._ctrl.channels.sort(key=lambda x: str(x.id[14:]))
+
+        for ch in self._ctrl._channels:
+            name = ch._id
+            self._rx_channel_names.append(name)
+            self.channel.append(self._channel(self._ctrl, name))
+        rx.__init__(self)
+        self.rx_buffer_size = 16
+
+    def rx(self):
```
Comment (tfcollins) [inline adi/adpd188.py:75]: You can just remove this method and use the built-in one for rx. I think this was just copy and pasted from an older device where the driver didn't scale the data correctly into scientific units

**Comment by adrimbarean** [comment]
V2: 
 - remove unnecessary rx() redfinition;

**Pull request merged**